### PR TITLE
Prevent starting new chat flow after reconnectUrl is used

### PR DESCRIPTION
**Pull request by charliewang95**
### Description
_Include a description of the problem to be solved_
When using auth chat, and a reconnectURL is provided, sometimes after clicking the chat button it starts a new chat (or shows prechat if it's configured)

## Solution Proposed
_Detail what is the solution proposed, include links to design document if required or any other document required to support the solution_
If reconnect has already kicked in, stop going with the normal startChat flow

### Acceptance criteria
_Define what are the conditions to consider the PR has achieved the intended goal_
After providing reconnectURL, the chat always reconnects to the same chat if it still exists, without starting new chat sor showing prechat again

## Test cases and evidence
_Include what tests cases were considered, any evidence of testing for future references, to identify any corner cases, etc_

[Test Case 3704158](https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3704158): [LCW OOB] [Reconnect Chat] Verify using reconnectUrl works on authenticated chats with prechat
[Test Case 3704160](https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3704160): [LCW OOB] [Reconnect Chat] Verify using reconnectUrl works on authenticated chats without prechat
[Test Case 3704162](https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3704162): [LCW OOB] [Reconnect Chat] Verify using reconnectUrl works on authenticated chats with prechat on new browser
[Test Case 3704163](https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3704163): [LCW OOB] [Reconnect Chat] Verify using reconnectUrl works on authenticated chats with prechat on new browser with duplicate tab

proof
proof2
proof3

### Sanity Tests
-   [ ] You have tested all changes in Popout mode
-   [ ] You have tested all changes in cross browsers i.e Edge, Chrome, Firefox, Safari and mobile devices(iOS and Android)
-   [ ] Your changes are included in the [CHANGELOG](../CHANGE_LOG.md)

## A11y
- [ ] You have run the [Automated Accessibility Check](https://accessibilityinsights.io/docs/en/windows/getstarted/automatedchecks) and have no reported issues

__*Please provide justification if any of the validations has been skipped.*__

### CODE REVIEW

**Review by sarojkpr — APPROVED** [review]

File: chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import { TelemetryHelper } from "../../../common/telemetry/TelemetryHelper";
 // Return value: should start normal chat flow when reconnect is enabled
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const handleChatReconnect = async (chatSDK: any, props: any, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, initStartChat: any, state: ILiveChatWidgetContext): Promise<boolean> => {
-    if (!isReconnectEnabled(props.chatConfig) || isPersistentEnabled(props.chatConfig)) return false;
+    if (!isReconnectEnabled(props.chatConfig) || isPersistentEnabled(props.chatConfig)) return true;
```
Comment (kevinlin98z) [inline chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts:24]: Were we leveraging the old `false` return value anywhere at all?

**Comment by kevinlin98z** [comment]
Why do we have different behavior of "waiting for the chat button to show up" for the "without prechat" and "directly loading the chat" for the "with prechat" scenarios?

**Comment by kevinlin98z** [comment]
Have we tested flows with unauthenticated chat + reconnectURL?

File: chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts (hunk 1 above)
Comment (charliewang95) [reply chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts:24]: It was previously protected by other logic 
image

**Comment by charliewang95** [comment]
> Have we tested flows with unauthenticated chat + reconnectURL?

Yes

**Comment by charliewang95** [comment]
> Why do we have different behavior of "waiting for the chat button to show up" for the "without prechat" and "directly loading the chat" for the "with prechat" scenarios?

If it's the same browser, due to local storage, the chat might rehydrate directly. If using new browser, the chat button shows up first. It has nothing to do with prechat or not. There might be some confusion in the test cases. Have fixed them

File: chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts (hunk 1 above)
Comment (charliewang95) [reply chat-widget/src/components/livechatwidget/common/reconnectChatHelper.ts]: Moved the check outside as suggested

**Review by xTEddie — APPROVED** [review]
